server.js

```diff
@@ -336,9 +337,14 @@ Server.prototype._onWebSocketRequest = function (socket, opts, params) {
     }
     if (self.destroyed) return
 
-    if (socket.infoHashes.indexOf(params.info_hash) === -1) {
-      socket.infoHashes.push(params.info_hash)
-    }
+    var hashes
+    if (typeof params.info_hash === 'string') hashes = [ params.info_hash ]
+    else hashes = params.info_hash
```

Comment:
Also, this change shouldn't be required because `socket.infoHashes` is only used so we can make sure that the client leaves all the swarms that they were a part of. See: https://github.com/feross/bittorrent-tracker/blob/7daf7ec2974f6a8de6b25ca01f75f8db82ce7f3e/server.js#L596

So, the proper change to make here is to only do this logic for announces, and skip it for scrapes.